baseplate/core.py

```diff
@@ -123,6 +124,145 @@ def from_upstream(cls, trace_id, parent_id, span_id, sampled, flags):
         return cls(trace_id, parent_id, span_id, sampled, flags)
 
 
+class AuthenticationContextFactory(object):
+    """Building factory for :py:class:`AuthenticationContext` objects, handles
+    context dependency for the secrets store.
+
+    This factory should be passed into the constructor of any upstream
+    :doc:`integrations <integration/index>` so that it is aware of the
+    application's secret store and can create the authentication context for
+    the incoming requests.
+
+    :param baseplate.secrets.SecretsStore secrets_store: the application's
+        defined secrets store, where application secret lookups should be made
+    """
+    def __init__(self, secrets_store=None):
+        self.secrets = secrets_store
+
+    def make_context(self, token):
+        """:py:class:`AuthenticationContext` builder
+
+        :param token: JWT value originating from the Authentication service
+            either directly or from an upstream service
+        :rtype: :py:class:`AuthenticationContext`
+        """
+        return AuthenticationContext(token=token, secrets=self.secrets)
+
+
+class AuthenticationContext(object):
+    """Wrapper for the contextual authentication information
+
+    :param str token: the JWT value for the currently propagated authentication
```

Comment:
The comment was mostly in reference to your second comment (about not even documenting because only the factory is building it).  It is also a concern for another subsequent comment about double defaulting.